browser/resources/settings/brave_sync_page/sync_url_input.ts

```diff
@@ -0,0 +1,180 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+/**
+ * @fileoverview `sync-url-input` is a single-line text field with title
+ *  that is used for setting up custom sync server url. It is based on
+ * `home-url-input`.
+ */
+import 'chrome://resources/cr_elements/cr_textarea/cr_textarea.js'
+import '/shared/settings/prefs/prefs.js'
+
+import type { CrInputElement } from 'chrome://resources/cr_elements/cr_input/cr_input.js'
+import type { CrPolicyPrefMixinInterface } from '/shared/settings/controls/cr_policy_pref_mixin.js'
+
+import { assert } from 'chrome://resources/js/assert.js'
+import { PolymerElement } from 'chrome://resources/polymer/v3_0/polymer/polymer_bundled.min.js'
+import { PrefControlMixin } from '/shared/settings/controls/pref_control_mixin.js'
+import { PrefsMixin } from '/shared/settings/prefs/prefs_mixin.js'
+import { CrPolicyPrefMixin } from '/shared/settings/controls/cr_policy_pref_mixin.js'
+import {
+  RelaunchMixin,
+  RelaunchMixinInterface,
+  RestartType
+} from '../relaunch_mixin.js'
+
+import { BraveSyncBrowserProxy } from './brave_sync_browser_proxy.js'
+
+import { getTemplate } from './sync_url_input.html.js'
+
+export interface SyncUrlInputElement {
+  $: {
+    input: CrInputElement
+  }
+}
+
+const SyncUrlInputElementBase = RelaunchMixin(
+  PrefsMixin(PrefControlMixin(CrPolicyPrefMixin(PolymerElement)))
+) as {
+  new (): PolymerElement & CrPolicyPrefMixinInterface & RelaunchMixinInterface
+}
+
+export class SyncUrlInputElement extends SyncUrlInputElementBase {
```

Comment:
why a custom input element is required for this?

Reply:
I think it needs a custom one because group policy can now override this value (see https://github.com/brave/brave-core/pull/25498) and the input should be read-only for that case.

Reply:
Yes! As  @bsclifton  mentioned we should make it read-only when group policy applied  for it and another reason is that  I couldn't find a common element to  validate and/or show relaunch when the input changes,  which has been a bit tricky for me to adapt using something similar to  `home-url-input`. So I felt it would be better to have a separate input to handle all the cases in one place.  Kindly let me know if you have any  further suggestions.
@goodov